File: people/server/src/main/java/ru/ifmo/se/dating/people/api/HttpPeopleApi.java
package ru.ifmo.se.dating.people.api;

import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.Set;
import java.util.UUID;

import javax.validation.constraints.Min;

import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import ru.ifmo.se.dating.people.model.generated.Faculty;
import ru.ifmo.se.dating.people.model.generated.Person;
import ru.ifmo.se.dating.people.model.generated.PersonDraft;
import ru.ifmo.se.dating.people.model.generated.ZodiacSign;

import ru.ifmo.se.dating.people.api.generated.PeopleApiDelegate;

@Service
class HttpPeopleApi implements PeopleApiDelegate {

    @SuppressWarnings("checkstyle:ParameterNumber")
    @Override
    public ResponseEntity<Set<Person>> peopleGet(
        Long offset, Long limit, String firstName, String lastName,
        Integer picturesCountMin, Integer picturesCountMax,
        List<@Min(1) Long> topicId, Integer heightMin,
        Integer heightMax, LocalDate birthdayMin, LocalDate birthdayMax,
        List<ZodiacSign> zodiac, List<Faculty> faculty, Double latitude,
        Double longitude, Integer radius, OffsetDateTime updatedMin,
        OffsetDateTime updatedMax, Set<String> sortBy
    ) {
        return ResponseEntityStub.create();
    }

    @Override
    public ResponseEntity<Void> peoplePersonIdDelete(Long personId) {
        return ResponseEntityStub.create();
    }

    @Override
    public ResponseEntity<Person> peoplePersonIdGet(Long personId) {
        return ResponseEntityStub.create();
    }

    @Override
    public ResponseEntity<Void> peoplePersonIdPatch(
        Long personId, PersonDraft personDraft) {
        return ResponseEntityStub.create();
    }

    @Override
    public ResponseEntity<Person> peoplePost(
        UUID idempotencyKey, PersonDraft personDraft) {
        return ResponseEntityStub.create();
    }
}

File: people/server/src/main/kotlin/ru/ifmo/se/dating/people/api/HttpPeopleApi.kt
package ru.ifmo.se.dating.people.api

import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import ru.ifmo.se.dating.people.api.generated.PeopleApiDelegate
import ru.ifmo.se.dating.people.model.generated.Faculty
import ru.ifmo.se.dating.people.model.generated.Person
import ru.ifmo.se.dating.people.model.generated.PersonDraft
import ru.ifmo.se.dating.people.model.generated.ZodiacSign
import java.time.LocalDate
import java.time.OffsetDateTime
import java.util.*

@Service
internal class HttpPeopleApi : PeopleApiDelegate {
    override fun peopleGet(
        offset: Long,
        limit: Long,
        firstName: String?,
        lastName: String?,
        picturesCountMin: Int?,
        picturesCountMax: Int?,
        topicId: List<Long>?,
        heightMin: Int?,
        heightMax: Int?,
        birthdayMin: LocalDate?,
        birthdayMax: LocalDate?,
        zodiac: List<ZodiacSign>?,
        faculty: List<Faculty>?,
        latitude: Double?,
        longitude: Double?,
        radius: Int?,
        updatedMin: OffsetDateTime?,
        updatedMax: OffsetDateTime?,
        sortBy: List<String>?,
    ): ResponseEntity<List<Person>> =
        ResponseEntityStub.create()

    override fun peoplePersonIdDelete(personId: Long): ResponseEntity<Unit> =
        ResponseEntityStub.create()

    override fun peoplePersonIdGet(personId: Long): ResponseEntity<Person> =
        ResponseEntityStub.create()

    override fun peoplePersonIdPatch(
        personId: Long,
        personDraft: PersonDraft,
    ): ResponseEntity<Unit> =
        ResponseEntityStub.create()

    override fun peoplePost(
        idempotencyKey: UUID,
        personDraft: PersonDraft,
    ): ResponseEntity<Person> =
        ResponseEntityStub.create()
}
